shopfloor/services/exception.py

```diff
@@ -0,0 +1,117 @@
+from functools import wraps
```

Comment:
and this is for?

Reply:
This was an attempt to find a better way to handle common error in the services. wraps is used to wraps the decorated function and allows us to keep the signature and name of the decorated function.